packages/alfa-web/src/page.ts

```diff
@@ -119,12 +121,18 @@ export namespace Page {
     "dct:hasPart": [Request.EARL, Response.EARL];
   }
 
-  export function from(json: JSON): Page {
-    return Page.of(
-      Request.from(json.request),
-      Response.from(json.response),
-      Document.from(json.document),
-      Device.from(json.device)
+  export function from(json: JSON): Result<Page, string> {
+    return Request.from(json.request).andThen((request) =>
+      Response.from(json.response).andThen((response) =>
+        Result.of(
```

Comment:
```suggestion
      Response.from(json.response).map((response) =>
```
Wouldn't that work also?